src/store/home/reducer.js

```diff
@@ -0,0 +1,41 @@
+import {
+  FETCH_ALL_POSTS_SUCCESS,
+  ADD_A_COMMENT,
+  DELETE_A_POST,
+} from "./actions";
+import { LIKE_A_POST_SUCCESS, UNLIKE_A_POST_SUCCESS } from "../user/actions";
+
+const initialState = {
+  posts: [],
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case FETCH_ALL_POSTS_SUCCESS:
+      return { ...state, posts: action.payload };
+
+    case LIKE_A_POST_SUCCESS:
+    case UNLIKE_A_POST_SUCCESS:
+      let index = state.posts.findIndex(
+        (post) => post.id === action.payload.id
+      );
+      state.posts[index] = action.payload;
+      return { ...state, ...(state.posts[index] = action.payload) };
```

Comment:
nope nope nope, no mutating the state! 😄 so meaning this: `state.posts[index] = action.payload;` is not allowed.
We can rewrite this like:
```
const newPostArray = state.posts.map(post => post.id === action.payload.id ? {...action.payload} : post)
return { ...state, posts: newPostArray }
```

Reply:
so we map over all the posts and then when we find a match we substitute the one we had in our state array with the updated version